File: src/game/inventory/gold.ts
import type { GameDatabase } from '../data/types'

/** Fallback currency item when Config has no `currency_item_id`. */
export const GOLD_ITEM_ID = 'ITEM-0001'

/** Gold item ID from Config, or [GOLD_ITEM_ID] when the row is missing. */
export function currencyItemId(db?: GameDatabase): string {
  if (!db) return GOLD_ITEM_ID
  const raw = db.Config.find((row) => row.Key === 'currency_item_id')?.Value
  return typeof raw === 'string' && raw.length > 0 ? raw : GOLD_ITEM_ID
}

/** Primary currency item — never stays in the bag; converts to `save.gold`. */
export function isGoldCurrencyItem(itemId: string, db?: GameDatabase): boolean {
  return itemId === currencyItemId(db)
}

File: packages/ik_rules/lib/src/inventory/gold.dart
import 'package:collection/collection.dart';
import 'package:ik_content/ik_content.dart';

/// Fallback currency item when Config has no `currency_item_id`.
const String goldItemId = 'ITEM-0001';

/// Gold item ID from Config, or [goldItemId] when the row is missing.
String currencyItemId([GameDatabase? db]) {
  if (db == null) return goldItemId;
  final raw = db.config.firstWhereOrNull((row) => row.raw['Key'] == 'currency_item_id')?.raw['Value'];
  return raw is String && raw.isNotEmpty ? raw : goldItemId;
}

/// Primary currency item — never stays in the bag; converts to `save.gold`.
bool isGoldCurrencyItem(String itemId, [GameDatabase? db]) => itemId == currencyItemId(db);
